### ISSUE TITLE
Configure Portal with a certificate from user's CA

### ISSUE THREAD

**Issue by jameshbarton** [solo-io/gloo-portal-issues#34]
**Is your feature request related to a problem? Please describe.**
A user has an IdP that requires certificates from a CA that Portal doesn't recognize out of the box.  This makes it difficult to integrate with these IdPs.

**Describe the solution you'd like**
Allow users to provide a secret where customers can specify a certificate from the user's CA.

**Describe alternatives you've considered**
A workaround is for the user to add the certificate to the portal container image.

**Comment by marcogschmidt** [comment]
A simple solution to this issue is to allow users to mount a CA secret to the controller container (via Helm values). We should then make sure that all the HTTPS requests that we send from the container use an [HTTP client](https://golang.org/pkg/net/http/#Client) configured with the mounted CA (see an example of how to do that [here](https://stackoverflow.com/questions/38582937/how-to-create-a-tls-client-with-ca-certificates-in-go)).

**Comment by sirrapa** [comment]
For now the workarround is using kustomize.

```
apiVersion: apps/v1
kind: Deployment
metadata:
  namespace: gloo-portal
  name: gloo-portal-admin-server
spec:
  template:
    spec:
      containers:
        - name: server
          volumeMounts:
            - mountPath: /etc/ssl/certs/
              name: certs
      initContainers:
        - name: add-ca-cert
          args:
            - -c
            - cp -r /etc/ssl/certs/* /certs; cat /etc/ssl/certs/ca-certificates.crt /ca-certs/ca.crt > /certs/ca-certificates.crt
          command:
            - sh
          image: gcr.io/gloo-portal/gloo-portal-admin-server:1.0.1
          imagePullPolicy: IfNotPresent
          volumeMounts:
            - mountPath: /certs
              name: certs
            - mountPath: /ca-certs
              name: ca-certs
      volumes:
        - emptyDir: {}
          name: certs
        - name: ca-certs
          secret:
            defaultMode: 420
            items:
              - key: custom-ca.crt
                path: ca.crt
            secretName: custom-ca-certs

**Comment by marcogschmidt** [comment]
Thanks for the info @sirrapa. An even better approach then the one I outlined above would be to update our Helm chart so that users can easily produce the configuration in your example. E.g. we could add a new `customCaSecretName` Helm value and render the deployment manifest based on its value.

**Comment by marcogschmidt** [comment]
This has been implemented and released with Gloo Portal `v1.1.0-beta3`.
